src/services/popover-orientation/popover-orientation.service.ts

```diff
@@ -60,7 +66,7 @@ export class PopoverOrientationListener {
 
     private onScrollOrResize() {
         this._rect = this._elementParent ? this._elementParent.parentElement.getBoundingClientRect() : this._element.parentElement.getBoundingClientRect();
-        const itemHeight = this._element.offsetHeight;
+        const itemHeight = this._element.offsetHeight || this._maxHeight;
```

Comment:
What is the situation that is being handled here? (Comments would be helpful.)

Reply:
If the element height is 0, then it will instead use the max height value. This stops the dropdown from briefly appearing in the wrong direction and then switching to the correct one

Reply:
Is this because of the animation? What problem would occur if maxHeight was not available in pixels?

Reply:
It seems to be because at this point the element may not actually exist and have a height of 0, so it'll incorrectly assume that there's enough space based on it having a height of 0 pixels. Then when the element is actually created it'll correct this and move to the up direction. I'm not sure how to prevent this if the value is not in pixels